core/trino-spi/src/main/java/io/trino/spi/exchange/ExchangeSourceSplitter.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.exchange;
+
+import java.io.Closeable;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public interface ExchangeSourceSplitter
+        extends Closeable
+{
+    CompletableFuture<?> NOT_BLOCKED = CompletableFuture.completedFuture(null);
+
+    CompletableFuture<?> isBlocked();
+
+    Optional<ExchangeSourceHandle> getNext();
```

Comment:
Why not use `hasNext` and `next`?

Reply:
Actually why do we need iterator here at all?
Maybe simpler interface with 
```
List<ExchangeSourceHandle> split();
```
would be enough?

Reply:
@linzebing 


In our interfaces usually we prefer to avoid having another `hasNext` method (I guess for both, styling and efficiency reasons). I'm trying to keep it in the same style as the other iterative interfaces.

@losipiuk 

You are right, with the current implementation `List<ExchangeSourceHandle> split();` should do the job. However in the future we may need this process to be iterative. For example if an Exchange implementation allows to start consuming earlier (before the producers finish) the split process might be ongoing. Since this is an interface in the SPI I was trying to future proof it.

Reply:
> However in the future we may need this process to be iterative

Let's keep it then. I am not super concerned about changing SPI. Yet it does not add too much complexity even in iterative form.